peps/pep-0733.rst

```diff
@@ -0,0 +1,634 @@
+details, because it exposes struct fields such as ``ob_refcnt``,
+``ob_type`` and ``ob_size``. There was some discussion about whether
+the stable ABI is worth keeping. Arguments on both sides can be
+found in [`Issue 4 <https://github.com/capi-workgroup/problems/issues/4>`__]
+and [`Issue 9 <https://github.com/capi-workgroup/problems/issues/9>`__].
+
+Alternatively, it was suggested that in order to be able to evolve
+the stable ABI, we need a mechanism to support multiple versions of
+it in the same Python binary. It was pointed out that versioning
+individual functions within a single ABI version is not enough
+because it may be necessary to evolve, together, a group of functions
+that interoperate with each other
+[`Issue 39 <https://github.com/capi-workgroup/problems/issues/39>`__].
+
+The limited API was introduced in 3.2 as a blessed subset of the C API
+which is recommended for users who would like to restrict themselves
+to high quality APIs which are not likely to change often. The
+``Py_LIMITED_API`` flag allows users to restrict their program to older
+versions of the limited API, but we now need the opposite option, to
+exclude older versions. This would make it possible to evolve the
+limited API by replacing flawed elements in it
+[`Issue 54 <https://github.com/capi-workgroup/problems/issues/54>`__].
+More generally, in a redesign we should revisit the way that API
+tiers are specified and consider designing a method that will unify the
+way we currently select between the different tiers
+[`Issue 59 <https://github.com/capi-workgroup/problems/issues/59>`__].
+
+API elements whose names begin with an underscore are considered
+private, essentially an API tier with no stability guarantees.
+However, this was only clarified recently, in :pep:`689`. It is
+not clear what the change policy should be with respect to such
+API elements that predate PEP 689
+[`Issue 58 <https://github.com/capi-workgroup/problems/issues/58>`__].
+
+There are API functions which have an unsafe (but fast) version as well as
+a safe version which performs error checking (for example,
+``PyTuple_GET_ITEM`` vs ``PyTuple_GetItem``). It may help to
+be able to group them into their own tiers - the "unsafe API" tier and
+the "safe API" tier
+[`Issue 61 <https://github.com/capi-workgroup/problems/issues/61>`__].
+
+Use of the C Language
+---------------------
+
+A number of issues were raised with respect to the way that CPython
+uses the C language. First there is the issue of which C dialect
+we use, and how we test our compatibility with it
+[`Issue 42 <https://github.com/capi-workgroup/problems/issues/42>`__].
+
+Usage of ``const`` in the API is currently sparse, but it is not
+clear whether this is something that we should consider changing
+[`Issue 38 <https://github.com/capi-workgroup/problems/issues/38>`__].
+
+We currently use the C types ``long`` and ``int``, where ``stdint``
+and ``int32_t`` would have been better choices
```

Comment:
I am not sure what `stdint` is supposed to mean here, perhaps you mean fixed-width integers in general?
```suggestion
We currently use the C types ``long`` and ``int``, where fixed-width integers
such as ``int32_t`` and ``int64_t`` would have been better choices
```